### DISCUSSION TITLE
Use of `unload` event

### DISCUSSION THREAD

**Discussion by yurtsiv** [ueberdosis/hocuspocus#740] [General]
Hocuposucs provider [uses](https://github.com/ueberdosis/hocuspocus/blob/74c53ccef08e81017b59ef7cf9131d4344867d1b/packages/provider/src/HocuspocusProvider.ts#L310) `unload` event which is [not very reliable](https://developer.chrome.com/articles/page-lifecycle-api/#the-unload-event).

Was it a deliberate choice or have you just not had a chance to try  `visibilitychange` as an alternative?

Thanks!

**Comment by janthurau** [comment]
hey @yurtsiv, thanks for raising this! I don't think it was a deliberate choice.

We're just using this to remove awareness, which, in case the event does not fire, will happen after the provider was absent for 30 seconds, so it's probably not a big issue. Won't hurt to switch to `visibilitychange`, so feel free to open a PR if you are interested in the change :)

**Comment by yurtsiv** [comment]
Fixed https://github.com/ueberdosis/hocuspocus/pull/824
